feature_engine/transformation/log.py

```diff
@@ -164,3 +164,206 @@ def _more_tags(self):
+
+    Attributes
+    ----------
+    variables_:
+        The group of variables that will be transformed.
+
+    n_features_in_:
+        The number of features in the train set used in fit.
+
+    C_:
+        The constant C to transform data.
+
+    Methods
+    -------
+    fit:
+        This transformer learns the constant C to add to log transformation.
+    transform:
+        Transforms the variables using log transformation.
+    fit_transform:
+        Fit to data, then transform it.
+    """
+
+    def __init__(
+        self,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+        base: str = "e",
+        C: Union[int, str] = "auto",
+    ) -> None:
+
+        if base not in ["e", "10"]:
+            raise ValueError("base can take only '10' or 'e' as values")
+
+        if not isinstance(C, int) and not C == "auto":
+            raise ValueError("C can take only 'auto' or int")
+
+        self.variables = _check_input_parameter_variables(variables)
+        self.base = base
+        self.C = C
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        This transformer learns the constant C to add to log transformation.
+
+        Select the numerical variables and determines whether the logarithm
+        can be applied on the selected variables (it checks if the variables
+        are all positive after the addition of C).
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shape = [n_samples, n_features].
+            The training input samples. Can be the entire dataframe, not just the
+            variables to transform.
+
+        y: pandas Series, default=None
+            It is not needed in this transformer. You can pass y or None.
+
+        Raises
+        ------
+        TypeError
+            - If the input is not a Pandas DataFrame
+            - If any of the user provided variables are not numerical
+        ValueError
+            - If there are no numerical variables in the df or the df is empty
+            - If the variable(s) contain null values
+            - If some variables contain zero or negative values after suming constant C
+
+        Returns
+        -------
+        self
+        """
+
+        # check input dataframe
+        X = super().fit(X)
+
+        # calculate C to add to each variable
+        if self.C == "auto":
+            self.C_ = X[self.variables_].min(axis=0).abs() + 1
+        else:
+            self.C_ = self.C
+
+        # check contains zero or negative values
+        if (X[self.variables_] + self.C_ <= 0).any().any():
+            raise ValueError(
+                "Some variables contain zero or negative values after suming"
+                + "constant C, can't apply log"
+            )
+
+        self.n_features_in_ = X.shape[1]
+
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Transforms the variables using log transformation.
```

Comment:
Transforms the variables using **the** log transformation. Or more directly "Transforms the variables with the logarithm" (probably better).

Probably also needs changing in the old logtransformer above.

Reply:
I've not seen this one, gonna make the change